### PR TITLE
Substantive updates to DID Method Requirements section.

### PR DESCRIPTION
**Pull request by msporny**
This is a big rework of the methods section. Almost all normative requirements were kept, but reworded slightly or reduced in size due to other overlapping normative statements. I've marked these changes as substantive because, although none of the changes would affect software implementations, some of the changes would affect what was expected to be written by DID Method Specification authors.

The only normative statement that I removed was "The <code>method-specific-id</code> value MUST be able to be generated without the use of a centralized registry service." -- which isn't true anymore, since we allow centralized registries to be used for DID Methods.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/pull/666.html" title="Last updated on Feb 19, 2021, 8:24 PM UTC (7e95c66)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/did-core/666/f768dd1...7e95c66.html" title="Last updated on Feb 19, 2021, 8:24 PM UTC (7e95c66)">Diff</a>

### CODE REVIEW

**Review by OR13 — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -2769,7 +2769,7 @@ <h3>Consumption</h3>
 
         <p>
 If media type information is available to a <a>conforming consumer</a> and the
-media type value is <code>application/did+json</code>: the data structure 
+media type value is <code>application/did+json</code>: the data structure
```
Comment (TallTed) [inline index.html:2772]:
```suggestion
media type value is <code>application/did+json</code>, then the data structure
```

File: index.html [hunk 2]
```diff
@@ -2769,7 +2769,7 @@ <h3>Consumption</h3>
...
 being consumed is a <a>DID document</a> and the root element MUST be a <a
```
Comment (TallTed) [inline index.html:2773]:
```suggestion
being consumed is a <a>DID document</a>, and the root element MUST be a <a
```

File: index.html [hunk 3]
```diff
@@ -3302,94 +3302,88 @@ <h3>CBOR Extensibility</h3>
...
-[[RFC7230]]). It is also similar to the relationship between the IETF generic
-URN specification ([[RFC8141]]) and a specific URN namespace definition (such as
-the <a>UUID</a> URN namespace defined in [[RFC4122]]). The difference is that a
-DID method specification, in addition to defining a specific <a>DID scheme</a>,
-also specifies the methods creating, resolving, updating, and deactivating
-<a>DIDs</a> and <a>DID documents</a> using a specific type of <a>verifiable data
-registry</a>.
+<a>URI</a> specification [[?RFC3986]] and a specific <a>URI</a> scheme
+[[?IANA-URI-SCHEMES]], such as the <code>http</code> scheme [[?RFC7230]].  In
+addition, to defining a specific <a>DID scheme</a>, a <a>DID method</a>
```
Comment (TallTed) [inline index.html:3317]:
```suggestion
addition to defining a specific <a>DID scheme</a>, a <a>DID method</a>
```

File: index.html [hunk 4]
```diff
@@ -3302,94 +3302,88 @@ <h3>CBOR Extensibility</h3>
...
+      <h2>Method Syntax</h2>
 
       <p>
-A <a>DID method</a> specification MUST define exactly one method-specific
-<a>DID scheme</a>, identified by exactly one method name (see the
-<code>method-name</code> rule in Section <a href="#did-syntax"></a>).
-      </p>
-
+The requirements for all <a>DID method</a> specifications when defining the
+method-specific DID Syntax are:
```
Comment (TallTed) [inline index.html:3335]:
```suggestion
method-specific DID Syntax are as follows:
```

File: index.html [hunk 5]
```diff
@@ -3302,94 +3302,88 @@ <h3>CBOR Extensibility</h3>
...
-      <p>
+        </li>
+        <li>
+The <a>DID method</a> specification MUST define sensitivity and normalization of
+the value of the <code>method-specific-id</code>.
+        </li>
+        <li>
+The <code>method-specific-id</code> value MUST be unique within a <a>DID
+method</a>. The <code>method-specific-id</code> value might be globally unique
+by itself.
```
Comment (TallTed) [inline index.html:3355]:
```suggestion
method</a>. The <code>method-specific-id</code> value itself might be globally
unique.
```

File: index.html [hunk 6]
```diff
@@ -3302,94 +3302,88 @@ <h3>CBOR Extensibility</h3>
...
+        </li>
+        <li>
+Any <a>DID</a> generated by a <a>DID method</a> MUST be globally unique.
+        </li>
+        <li>
+In order to reduce the chances of <code>method-name</code> conflicts, a <a>DID
+method</a> specification SHOULD be registered in the DID Specification
+Registries [[?DID-SPEC-REGISTRIES]].
```
Comment (TallTed) [inline index.html:3363]:
```suggestion
To reduce the chances of <code>method-name</code> conflicts, a <a>DID method</a>
specification SHOULD be registered in the DID Specification Registries
[[?DID-SPEC-REGISTRIES]].
```

File: index.html [hunk 7]
```diff
@@ -3398,92 +3398,62 @@ <h2>Method Syntax</h2>
 
     <section>
       <h2>Method Operations</h2>
+
       <p>
-This section sets out the requirements for <a>DID method</a> specifications
-with regards to operations that can be performed on a <a>DID document</a>.
-      </p>
+The requirements for all <a>DID method</a> specifications when defining the
+method operations are:
```
Comment (TallTed) [inline index.html:3404]:
```suggestion
method operations are as follows:
```

File: index.html [hunk 8]
```diff
@@ -3398,92 +3398,62 @@ <h2>Method Syntax</h2>
...
+deactivate a <a>DID</a> <em>or</em> state that deactivation is not possible.
+        </li>
+      </ol>
+
       <p>
-Determining the authority of a party to carry out the operations is
-method-specific. For example, a <a>DID method</a> might:
+The authority of a party that is performing authorization to carry out the
+operations is specific to a <a>DID method</a>. For example, a <a>DID method</a>
+might:
```
Comment (TallTed) [inline index.html:3435]:
```suggestion
might &mdash;
```

File: index.html [hunk 9]
```diff
@@ -3398,92 +3398,62 @@ <h2>Method Syntax</h2>
...
-Document</a>.
+use other constructs in the <a>DID Document</a> such as the
+<a>verification method</a> specified via the <code><a>capabilityInvocation</a></code>
+<a>verification relationship</a>.
         </li>
         <li>
-not use the <a>DID document</a> at all to decide this, but have rules that are
-"built into" the method.
+not use the <a>DID document</a> at all to decide this and depend, instead, on an
+out-of-band mechanism.
```
Comment (TallTed) [inline index.html:3453]:
```suggestion
not use the <a>DID document</a> for this decision at all, and depend on an
out-of-band mechanism, instead.
```

File: index.html [hunk 10]
```diff
@@ -3458,70 +3458,79 @@ <h2>Method Operations</h2>
...
-      <p>
-If the technology involves authentication, particularly user-host
-authentication, the security of the authentication method MUST be clearly
-specified.
-      </p>
-      <p>
-<a>DID methods</a> MUST discuss the policy mechanism by which <a>DIDs</a> are proven to be uniquely assigned.
-      </p>
+The requirements for all <a>DID method</a> specifications when authoring the
+<em>Security Considerations</em> section are:
```
Comment (TallTed) [inline index.html:3464]:
```suggestion
<em>Security Considerations</em> section are as follows:
```

File: index.html [hunk 11]
```diff
@@ -3458,70 +3458,79 @@ <h2>Method Operations</h2>
...
+        <li>
+A <a>DID method</a> specifications MUST follow all guidelines and normative
+language provided in <a data-cite="RFC3552#section-5">RFC3552: Writing Security
+Considerations Sections</a> for the <a>DID</a> operations defined in the <a>DID
+method</a> specification.
+        </li>
+        <li>
+At least the following forms of attack MUST be documented for the <a>DID</a>
+operations defined in the <a>DID method</a> specification: eavesdropping,
+replay, message insertion, deletion, modification, denial of service,  storage
```
Comment (TallTed) [inline index.html:3477]:
```suggestion
replay, message insertion, deletion, modification, denial of service, storage
```

File: index.html [hunk 12]
```diff
@@ -3458,70 +3458,79 @@ <h2>Method Operations</h2>
...
+or network amplification, and man-in-the-middle.
```
Comment (TallTed) [inline index.html:3478]:
```suggestion
The Security Considerations section MUST document the following forms of attack
for the <a>DID</a> operations defined in the <a>DID method</a> specification:
eavesdropping, replay, message insertion, deletion, modification, denial of
service, storage or network amplification, and man-in-the-middle. Other known
forms of attack SHOULD also be documented.
```

Comment (TallTed) [reply index.html:3478]: Yeah, I added a normative statement... I think it is in the spirit of the section, and I can't think of any other way to say it that doesn't drastically weaken what *should* be a requirement.

File: index.html [hunk 13]
```diff
@@ -3458,70 +3458,79 @@ <h2>Method Operations</h2>
...
-on.
-      </p>
-      <p>
+        </li>
+        <li>
+If a protocol incorporates cryptographic protection mechanisms, the <a>DID
+method</a> specification MUST clearly indicate which portions of the data are
+protected, what the protections are, and SHOULD give an indication to what
+sorts of attacks the cryptographic protection is susceptible. For example,
+integrity only, confidentiality, or endpoint authentication.
```
Comment (TallTed) [inline index.html:3513]:
```suggestion
protected and by what protections, and it SHOULD give an indication of the
sorts of attacks to which the cryptographic protection is susceptible. Some
examples are integrity only, confidentiality, and endpoint authentication.
```

File: index.html [hunk 14]
```diff
@@ -3458,70 +3458,79 @@ <h2>Method Operations</h2>
...
+        <li>
 <a>DID method</a> specifications SHOULD explain and specify the implementation
 of signatures on <a>DID documents</a>, if applicable.
-      </p>
-      <p>
+        </li>
+        <li>
 Where <a>DID methods</a> make use of peer-to-peer computing resources, such as
 with all known <a>DLTs</a>, the expected burdens of those resources SHOULD be
 discussed in relation to denial of service.
```
Comment (TallTed) [inline index.html:3526]:
```suggestion
Where <a>DID methods</a> use peer-to-peer computing resources, such as with all
known <a>DLTs</a>, the expected burdens of those resources SHOULD be discussed
in relation to denial of service.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
A bunch of minor adjustments, a couple not-so-minor...

**Comment by msporny** [comment]
Substantive, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**